src/pages/integrations/analytics/custom-events/index.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Custom events for Beacon analytics
+---
+
+# Creating custom events
+
+PWA Studio 12.5 contains an eventing framework with a series of default events that cover common scenarios.
+By creating and importing a custom extension, it is possible to add new events so that you may track specific user interactions beyond what is included in the default package.
+
+PWA Studio provides a [sample extension](https://github.com/magento/pwa-studio/tree/develop/packages/extensions/experience-platform-connector/) that shows how to add a custom event.
+
+In the sample extension, the `SEARCH_REPSONSE` event is being tracked. This event is fired when a search result is returned from the Adobe Experience Manager (AEM), with either results or a `EMPTY_RESULT` response.
+
+These files can be used as a starting point for a custom event extension.
+
+## searchResponseReceived.js
+
+This file sets the payload and the search results context. The custom event name is also defined in this file.
```

Comment:
Seems to be a lot of relevant code missing from this snippet.

Also, I would describe this file as an event handler file that returns two functions:

1. `canHandle()` - a function that receives an event from PWA Studio's eventing framework and returns true if the `handle()` function in this file should handle the incoming event
2. `handle()` - function that handles the event